metadata-ingestion/src/datahub/utilities/yaml_sync_utils.py

```diff
@@ -0,0 +1,35 @@
+import contextlib
+import pathlib
+from typing import Any, Iterator
+
+import ruamel.yaml.util
+from ruamel.yaml import YAML
+
+
+@contextlib.contextmanager
+def YamlFileUpdater(file: pathlib.Path) -> Iterator[Any]:
+    yaml = YAML()
+    yaml.preserve_quotes = True  # type: ignore[assignment]
+
+    doc = yaml.load(file)
```

Comment:
Why did they make this stateful YAML object the way of doing this lol. I am not a fan